csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -0,0 +1,209 @@
+﻿/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Apache.Arrow.Ipc;
+using Apache.Hive.Service.Rpc.Thrift;
+using Thrift.Protocol;
+using Thrift.Transport;
+
+namespace Apache.Arrow.Adbc.Drivers.Apache.Hive2
+{
+    public abstract class HiveServer2Connection : AdbcConnection
+    {
+        const string userAgent = "AdbcExperimental/0.0";
+
+        protected TOperationHandle operationHandle;
+        protected IReadOnlyDictionary<string, string> properties;
+        internal TTransport transport;
+        internal TCLIService.Client client;
+        internal TSessionHandle sessionHandle;
+
+        internal HiveServer2Connection() : this(null)
+        {
+
+        }
+
+        internal HiveServer2Connection(IReadOnlyDictionary<string, string> properties)
+        {
+            this.properties = properties;
+        }
+
+        public void Open()
+        {
+            TProtocol protocol = CreateProtocol();
+            this.transport = protocol.Transport;
+            this.client = new TCLIService.Client(protocol);
+
+            var s0 = this.client.OpenSession(CreateSessionRequest()).Result;
+            this.sessionHandle = s0.SessionHandle;
+        }
+
+        protected abstract TProtocol CreateProtocol();
+        protected abstract TOpenSessionReq CreateSessionRequest();
+
+        public override IArrowArrayStream GetObjects(GetObjectsDepth depth, string catalogPattern, string dbSchemaPattern, string tableNamePattern, List<string> tableTypes, string columnNamePattern)
```

Comment:
we can't remove it. HiveServer2 is the base class that Spark and Impala build on, but I will add details to the readme.

Reply:
I mean the body, or at least the parts of the body that aren't implemented. It could always throw a NotImplementedException for when e.g. depth != GetObjectsDepth.All.

Reply:
Added NotImplementedExceptions for unsupported depth values.
Added documentation to README.

Reply:
Actually, removed the implementation here. It was not work or useful as base functionality.